ethpm/package.py

```diff
@@ -72,6 +86,18 @@ def __repr__(self):
         version = self.version
         return "<Package {0}=={1}>".format(name, version)
 
+    @classmethod
+    def from_file(cls, file_path_or_obj):
+        if isinstance(file_path_or_obj, str):
+            with open(file_path_or_obj) as file_obj:
+                package_data = _load_package_data_from_file(file_obj)
+        elif hasattr(file_path_or_obj, 'read') and callable(file_path_or_obj.read):
+            package_data = _load_package_data_from_file(file_path_or_obj)
+        else:
+            assert False
```

Comment:
I believe that this case should raise a `TypeError` with a message something like:

> `The `Package.from_file` method takes either a filesystem path or a file-like object.  Got {0} instead`

Fill `{0}` in with `type(file_path_or_obj)`

Reply:
Got it 👍